src/transformers/trainer.py

```diff
@@ -1871,10 +1871,6 @@ def _inner_training_loop(
                         # deepspeed does its own clipping
 
                         if self.do_grad_scaling:
-                            # Reduce gradients first for XLA
-                            if is_torch_tpu_available():
-                                gradients = xm._fetch_gradients(self.optimizer)
-                                xm.all_reduce("sum", gradients, scale=1.0 / xm.xrt_world_size())
                             # AMP: gradients need unscaling
                             self.scaler.unscale_(self.optimizer)
```

Comment:
I feel like there's a bit more to this actually..... Because what we really need to have happen is any calls to `self.scaler.unscale_` should really now use `accelerator.unscale_gradients(self.optimizer)`.

Otherwise we won't be unscaling the gradients at all here. I know that's a bit more than the initial PR and will take a bit more work, but does this make sense? (This is the chunk of code where you put our `self.reduce` in the Accelerate PR). So let's try and do that as part of this PR if possible as well please :)

Reply:
> I feel like there's a bit more to this actually..... Because what we really need to have happen is any calls to self.scaler.unscale_ should really now use accelerator.unscale_gradients(self.optimizer).

It sounds good to me. It seems that there are some specific details mentioned here, which might need to be discussed further. If I'm not mistaken, `accelerator.unscale_gradients` is implicitly called when calling `self.accelerator.clip_grad_norm_`

https://github.com/huggingface/transformers/blob/fb7d246951d5f60aa36a7958841dfea72f51fc6b/src/transformers/trainer.py#L1931-L1935

I think the logic of `self.scaler.unsacle_(self.optimizer)` here only applies to ShardedDDP as we can see `self.do_grad_scaling` and `self.scaler` are initialized in the following code block
https://github.com/huggingface/transformers/blob/fb7d246951d5f60aa36a7958841dfea72f51fc6b/src/transformers/trainer.py#L629-L640

In order to be compatible with ShardedDDP, the current implementation is somewhat fragmented, so removing the ShardedDDP-related code first may help clarify things. WDYT @muellerzr

Reply:
Yes, especially as Accelerate uses `ShardedGradScaler` https://github.com/huggingface/accelerate/blob/main/src/accelerate/accelerator.py#L427-L434

Reply:
I apologize for not communicating my thoughts with enough precision. In transformers, `ShardedGradScaler` is imported from `from fairscale.optim.grad_scaler import ShardedGradScaler` not `from torch.distributed.fsdp.sharded_grad_scaler import ShardedGradScaler` @muellerzr

Reply:
Aren't we removing this anyways via https://github.com/huggingface/transformers/pull/25702?

Reply:
Yes, this PR should closed